infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/InfrastructureProvisioner.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.workspace.infrastructure.openshift;
+
+import org.eclipse.che.api.core.model.workspace.config.Environment;
+import org.eclipse.che.api.core.model.workspace.runtime.RuntimeIdentity;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.workspace.infrastructure.openshift.environment.OpenShiftEnvironment;
+
+/**
+ * Modifies environment config and OpenShift environment with infrastructure needs.
+ *
+ * @author Anton Korneta
+ */
+public interface InfrastructureProvisioner {
+
+    /**
+     * @param environment
+     *         configuration of environment
+     * @param osEnv
+     *         OpenShiftEnvironment
+     * @param identity
+     *         runtime identity
+     * @throws InfrastructureException
+     *         when any error occurs
+     */
+    void provision(Environment environment,
```

Comment:
I think EnvironmentImpl should be used instead since it can be changed according to javadocs